File: solutions/typescript/src/main4.ts
import * as fs from 'fs'

const inputs = fs.readFileSync('/dev/stdin', 'utf-8')
const a = inputs
  .split('\n')[1]
  .trim()
  .split(' ')
  .map((i) => Number(i))

const answer = Math.min(
  ...a.map((i) => {
    let count = 0
    while (i % 2 === 0) {
      count++
      i /= 2
    }
    return count
  })
)
console.log(answer)

File: solutions/rust/main4.rs
use proconio::input;

fn count(n: u32) -> u32 {
    if n % 2 != 0 {
        return 0;
    }
    return 1 + count(n / 2);
}

fn main() {
    input! {
        n: u32,
        a: [u32; n],
    }

    let answer = a.iter().map(|v| count(*v)).min().unwrap();
    println!("{}", answer);
}
